network/fabrics_test.go

```diff
@@ -0,0 +1,875 @@
+package network_test
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+
+	"strconv"
+
+	"io/ioutil"
+
+	"github.com/joyent/triton-go/network"
+	"github.com/joyent/triton-go/testutils"
+	"github.com/pkg/errors"
+)
+
+var (
+	fakeFabricVlanId      = 2
+	deleteFabricErrorType = errors.New("unable to delete fabric")
+	deleteVLanErrorType   = errors.New("unable to delete vlan")
+	createVLanErrorType   = errors.New("unable to create vlan")
+	createFabricErrorType = errors.New("unable to create fabric")
+	getVlanErrorType      = errors.New("unable to get vlan")
+	getFabricErrorType    = errors.New("unable to get fabric")
+	updateVlanErrorType   = errors.New("unable to update vlan")
+	listVlansErrorType    = errors.New("unable to list vlans")
+	listFabricsErrorType  = errors.New("unable to list fabrics")
+)
+
+func TestDeleteFabric(t *testing.T) {
+	networkClient := MockNetworkClient()
+
+	do := func(ctx context.Context, nc *network.NetworkClient) error {
+		defer testutils.DeactivateClient()
```

Comment:
I wish there were some symmetry to this `DeactivateClient()` call.  I see it being passed in to `do()`, but there's no concept of "acquiring" the resource, either in the form of a refcount incr or some kind of copy or other RAII idiom.  Is `DeactivateClient()` just a poorly named version of `Close()` for a mock'ed response?

And to be clear, I know this isn't specific to this PR so fixing this can be addressed in a follow up PR.